### ISSUE TITLE
Failing to run ExpHydro Model Tutorial

### ISSUE THREAD

**Issue by olivierbonte** [chooron/HydroModels.jl#4]
When running the tutorial as defined in https://github.com/chooron/HydroModels.jl/blob/35ff1bbd24172db898c701da079d8bbbbd9d2bc7/docs/src/tutorials/run_a_exphydro_model.md, I run into a few issues

- The input data should be mapped to an input matrix, as in https://github.com/chooron/HydroModels.jl/blob/35ff1bbd24172db898c701da079d8bbbbd9d2bc7/test/run_lumped_model.jl#L11
- The object `model` is not defined. To get this running, one should first follow the building Exphydro tutorial (https://github.com/chooron/HydroModels.jl/blob/35ff1bbd24172db898c701da079d8bbbbd9d2bc7/docs/src/implements/build_exphydro_model_en.md) and then construct the model using `model = HydroModel(; name=:exphydro, components=[snowpack_bucket,soilwater_bucket])`
- Converting the output to a named tuple as defined in https://github.com/chooron/HydroModels.jl/blob/35ff1bbd24172db898c701da079d8bbbbd9d2bc7/docs/src/tutorials/run_a_exphydro_model.md?plain=1#L88 does not work for me (using v0.1.1.) , although the code does run when setting `convert_to_ntp=true`.

**Comment by chooron** [comment]
This is quite embarrassing. This issue is also related to the documentation synchronization problem after code updates. In v0.1.1's design, I set the input and output restrictions for components like bucket and model to be AbstractArray type (the convert_to_ntp setting was therefore removed). If you want to use NamedTuple as input like before, you can wrapper this component through NamedTupleIOAdapter.

**Comment by olivierbonte** [comment]
No worries, I think these sort of issues are very normal when in early stages of package development! The updates you made to the docs in https://github.com/chooron/HydroModels.jl/commit/7727d7dbe0d9c6fb93d6f01da19533aa939222c0 make things a lot clearer already :) The only mistake I can still spot now is that `input` is not defined from the dataframe yet for use in 
https://github.com/chooron/HydroModels.jl/blob/7727d7dbe0d9c6fb93d6f01da19533aa939222c0/docs/src/tutorials/run_exphydro_model.md?plain=1#L100

This could be solved by adding
```
input_ntp = (lday=df[ts, "dayl(day)"], temp=df[ts, "tmean(C)"], prcp=df[ts, "prcp(mm/day)"])
input = Matrix(reduce(hcat, collect(input_ntp[[:temp, :lday, :prcp]]))')
```
somewhere before

**Comment by olivierbonte** [comment]
See PR #6

**Comment by olivierbonte** [comment]
Closed by PR #6
